core/Rename.hpp

```diff
@@ -56,13 +56,27 @@ namespace olympia
         //! \brief Name of this resource. Required by sparta::UnitFactory
         static const char name[];
 
+        // map of ARF -> PRF
+        std::queue<unsigned int> map_table[2][sparta::Scoreboard::MAX_REGISTERS];
+        // table used for storing PRF -> ARF mapping, so we can pop the map_table entries on retirement
+        std::queue<unsigned int> reverse_map_table[2][sparta::Scoreboard::MAX_REGISTERS];
+        // mapping used when a src doesn't reference a PRF
+        // we need to track those, so that when we retire the instruction, we know not to count them
+        // towards the reference counter
+        std::queue<unsigned int> src_map_table[2][sparta::Scoreboard::MAX_REGISTERS];
+        // reference counter for PRF
+        int reference_counter[2][sparta::Scoreboard::MAX_REGISTERS] = {};
+        // list of free PRF that are available to map
+        std::queue<unsigned int> freelist[2];
```

Comment:
Do not use a hard-coded number (2).  This should be based on the number of register files.

Reply:
Addressed in [fbe7264](https://github.com/riscv-software-src/riscv-perf-model/pull/40/commits/fbe7264a97b4aadea92369bdf93a544c3077551e)